File: Web/RMA-Web-Shell/projects/medicare/src/app/medical-invoice-manager/models/invoice-query-details.ts
import { BaseClass } from "projects/shared-models-lib/src/lib/common/base-class";
import { RolePlayerItemQueryTypeEnum, RolePlayerQueryItemTypeEnum, RolePlayerItemQueryCategoryEnum, 
    RolePlayerItemQueryStatusEnum } from "projects/shared-models-lib/src/lib/enums/roleplayer-item-query-enums";

export class InvoiceQueryDetails extends BaseClass {
    queryReferenceNumber: string;
    rolePlayerItemQueryType: RolePlayerItemQueryTypeEnum;
    rolePlayerId: number;
    rolePlayerQueryItemType: RolePlayerQueryItemTypeEnum;
    rolePlayerItemQueryCategory: RolePlayerItemQueryCategoryEnum;
    rolePlayerItemQueryStatus: RolePlayerItemQueryStatusEnum;
    queryDescription: string;
    queryResponse: string;
}

File: Service/MediCare/RMA.Service.MediCare.Contracts/Entities/Medical/InvoiceQueryDetails.cs
using RMA.Common.Entities;
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;

namespace RMA.Service.MediCare.Contracts.Entities.Medical
{
    public class InvoiceQueryDetails: AuditDetails
    {
        public RolePlayerItemQueryTypeEnum RolePlayerItemQueryType { get; set; }
        public int RolePlayerId { get; set; }
        public RolePlayerQueryItemTypeEnum RolePlayerQueryItemType { get; set; }
        public RolePlayerItemQueryCategoryEnum RolePlayerItemQueryCategory { get; set; }
        public RolePlayerItemQueryStatusEnum RolePlayerItemQueryStatus { get; set; }
        public string QueryDescription { get; set; }
        public string QueryResponse { get; set; }
        public string QueryReferenceNumber { get; set; }
    }
}
